ccmlib/node.py

```diff
@@ -1418,11 +1539,69 @@ def drain(self, block_on_log=False):
-        args = ["repair"] + options
-        cmd = ' '.join(args)
+    @property
+    def scylla_yaml(self) -> Dict[str, Any]:
+        return YAML().load(Path(self.get_conf_dir()) / common.SCYLLA_CONF)
+
+    @property
+    def api_port(self) -> int:
+        return self.scylla_yaml.get('api_port', 10000)
+
+    def cluster_command_available(self):
+        stdout, _ = self.nodetool('help')
+        return "  cluster  " in stdout
+
+    def run_repair(self, options, **kwargs):
+        cmd = "repair " + options
+        return self.nodetool(cmd, **kwargs)
+
+    def run_cluster_repair(self, options, **kwargs):
+        cmd = "cluster repair " + options
         return self.nodetool(cmd, **kwargs)
 
+    def repair(self, options=[], replication: RepairOptions.KeyspaceReplication = RepairOptions.KeyspaceReplication.ALL, keyspace: Optional[str] = None,
+               tables: Optional[list[str]] = None, hosts: Optional[list[str]] = None, dcs: Optional[list[str]] = None, dc_parallel: bool = False,
+               end_token: Optional[int] = None, full: bool = False, local: bool = False, ignore_unreplicated_keyspaces: bool = False,
+               job_threads: Optional[int] = None, partitioner_range: bool = False, pull: bool = False, sequential: bool = False, start_token: Optional[int] = None,
+               trace: bool = False, tablet_tokens: Optional[list[str]] = None, **kwargs):
+        if not options:
+            repair_options = RepairOptions(replication, keyspace, tables, hosts, dcs, dc_parallel, end_token, full, local, ignore_unreplicated_keyspaces,
+                                           job_threads, partitioner_range, pull, sequential, start_token, trace, tablet_tokens)
+            outs_and_errs = []
+            if repair_options.keyspace:
+                url = f"http://{self.address()}:{self.api_port}/storage_service/keyspaces"
+                resp = requests.get(url=url, params={"replication": "vnodes"})
+
+                if self.cluster_command_available() and resp.status_code == requests.codes.ok:
+                    if repair_options.keyspace in resp.json():
+                        res = self.run_repair(repair_options.get_vnode_repair_options(), **kwargs)
+                        outs_and_errs.append(res)
+                    else:
+                        res = self.run_cluster_repair(repair_options.get_tablet_repair_options(), **kwargs)
+                        outs_and_errs.append(res)
+                else:
+                    res = self.run_repair(repair_options.get_vnode_repair_options(), **kwargs)
+                    outs_and_errs.append(res)
+            else:
+                if self.cluster_command_available():
+                    if repair_options.repair_vnodes():
+                        res = self.run_repair(repair_options.get_vnode_repair_options(), **kwargs)
+                        outs_and_errs.append(res)
+
+                    if repair_options.repair_tablets() and self.cluster_command_available():
```

Comment:
I think we do not need to support the replication parameter to limit the scope of the repair to tablet or vnode. This is not useful in a test.